app/src/main/java/com/example/srv_twry/studentcompanion/Network/FetchContestsRetrofit.java

```diff
@@ -3,57 +3,68 @@
 import android.content.ContentValues;
 import android.content.Context;
 import android.net.Uri;
+import android.support.annotation.NonNull;
 import android.widget.Toast;
 
-import com.android.volley.Request;
-import com.android.volley.RequestQueue;
-import com.android.volley.Response;
-import com.android.volley.VolleyError;
-import com.android.volley.toolbox.JsonObjectRequest;
-import com.android.volley.toolbox.Volley;
 import com.example.srv_twry.studentcompanion.Database.DatabaseContract;
+import com.example.srv_twry.studentcompanion.Network.NetworkPOJOs.FeedPOJO;
+import com.example.srv_twry.studentcompanion.Network.NetworkPOJOs.ModelsPOJO;
 import com.example.srv_twry.studentcompanion.R;
 
-import org.json.JSONArray;
-import org.json.JSONException;
-import org.json.JSONObject;
+
+import java.util.List;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+import retrofit2.http.GET;
 
 /**
  * Created by srv_twry on 19/6/17.
  * The class which uses the volley library to fetch Contests from the Hackerrank's API
  */
 
-public class FetchContestsVolley {
-    private static RequestQueue requestQueue;
-    private static final String HACKERRANK_API_URL = "https://www.hackerrank.com/calendar/feed.json";
+public class FetchContestsRetrofit {
+    private static final String HACKERRANK_API_BASE_URL = "https://www.hackerrank.com";
+    private static final String CALENDAR_FEED_URL = "calendar/feed.json";
     private final Context context;
     private final onLoadingFinishedListener onLoadingFinishedListener;
+    private HackerrankService api;
 
-    public FetchContestsVolley(Context context,onLoadingFinishedListener onLoadingFinishedListener){
+    public FetchContestsRetrofit(Context context, onLoadingFinishedListener onLoadingFinishedListener){
         this.context=context;
         this.onLoadingFinishedListener= onLoadingFinishedListener;
-        requestQueue = Volley.newRequestQueue(context);
+
+        Retrofit retrofit = new Retrofit.Builder()
+                .baseUrl(HACKERRANK_API_BASE_URL)
+                .addConverterFactory(GsonConverterFactory.create())
+                .build();
+
+        api = retrofit.create(HackerrankService.class);
     }
 
-    public  void fetchContest(){
-        JsonObjectRequest jsonObjectRequest = new JsonObjectRequest(Request.Method.GET, HACKERRANK_API_URL, null, new Response.Listener<JSONObject>() {
+    public void fetchContest(){
+
+        api.getFeed().enqueue(new Callback<ModelsPOJO>() {
             @Override
-            public void onResponse(JSONObject response) {
-                addToDatabase(response);
+            public void onResponse(@NonNull Call<ModelsPOJO> call, @NonNull Response<ModelsPOJO> response) {
+
```

Comment:
Don't add unnecessary lines.

Reply:
i will correct the changes , but , doesn't this add readability to the code base ?